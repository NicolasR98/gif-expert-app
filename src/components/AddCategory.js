import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');

	const handleInputChange = ({ target: { value: key } }) => setInputValue(key);

	const handleAdd = (value) => setCategories((cats) => [value, ...cats]);

	const handleSubmit = (e) => {
		e.preventDefault();

		const isValidInput = inputValue.trim().length > 2;

		if (isValidInput) {
			handleAdd(inputValue);
			setInputValue('');
		}
	};

	return (
		<form onSubmit={handleSubmit}>
			<h2>Add Category</h2>
			<input type="text" value={inputValue} onChange={handleInputChange} />
		</form>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};
