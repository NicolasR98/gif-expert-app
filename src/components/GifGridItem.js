import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, id, url }) => {
	return (
		<div className="card  animate__animated animate__backInRight" key={id}>
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	id: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
	]).isRequired,
	url: PropTypes.string.isRequired,
};
