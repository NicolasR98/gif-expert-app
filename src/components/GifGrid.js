import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
	const { data: gifs, loading } = useFetchGifs(category);

	return (
		<div>
			<h3>{category}</h3>
			{loading && <p className="animate__animated animate__flash">Loading...</p>}
			<ol className="card-container">
				{gifs.map((gif) => (
					<GifGridItem key={gif.id} {...gif} />
				))}
			</ol>
		</div>
	);
};
