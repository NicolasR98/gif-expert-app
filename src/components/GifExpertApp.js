import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {
	const [categories, setCategories] = useState(['Age of empires 2']);

	return (
		<>
			<h2>GifExpertApp</h2>
			<hr />
			<AddCategory setCategories={setCategories} />
			<ol>
				{categories.map((category) => (
					<GifGrid key={category} category={category} />
				))}
			</ol>
		</>
	);
};
