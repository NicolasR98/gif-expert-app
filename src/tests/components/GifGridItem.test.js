import { shallow } from 'enzyme';
import react from 'react';
import { GifGridItem } from '../../components/GifGridItem';

describe('Tests on <GifGridItem />', () => {
	test('should show the component correctly', () => {
		const wrapper = shallow(<GifGridItem />);
		expect(wrapper).toMatchSnapshot();
	});
});
