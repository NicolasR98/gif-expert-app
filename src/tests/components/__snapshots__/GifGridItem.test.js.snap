// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem /> should show the component correctly 1`] = `
<div
  className="card  animate__animated animate__backInRight"
>
  <img />
  <p />
</div>
`;
